core/src/test/scala/unit/kafka/server/MetadataRequestTest.scala

```diff
@@ -52,7 +52,7 @@ class MetadataRequestTest extends BaseRequestTest {
     assertNotEquals("Controller id should switch to a new broker", controllerId, controllerId2)
     TestUtils.waitUntilTrue(() => {
       val metadataResponse2 = sendMetadataRequest(MetadataRequest.allTopics(), 1)
-      controllerServer2.apis.brokerId == metadataResponse2.controller.id
+      metadataResponse2.controller() != null && controllerServer2.apis.brokerId == metadataResponse2.controller.id
```

Comment:
Nitpick: no `()` needed after `controller`. For my benefit, can you please explain why the change in `KafkaServer` could cause the controller to be `null` in this case?